### PR TITLE
Python: several fixes

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
@globin @fpletz could we have a Hydra job for this PR or a new branch (say python-wip),  building the `pkgs/top-level/python-packages.nix` set? I would like to get all these changes in 17.09.

**Comment by jluttine** [comment]
How can I make a PR to this PR? I rebased my PR onto this PR, but when creating a PR, I don't know how to set this PR as the base..

**Comment by jluttine** [comment]
I did a PR onto FRidh:python-fixes.

**Comment by FRidh** [comment]
Rebased and added. Thanks @jluttine

**Comment by peti** [comment]
This PR introduces evaluation errors into Nixpkgs: 

~~~
$ nix-env -qaP --xml --meta 2>&1 >/dev/null
derivation ‘python2.7-pytest-warnings-0.3.0’ has invalid meta attribute ‘homepage’
derivation ‘python3.6-pytest-warnings-0.3.0’ has invalid meta attribute ‘homepage’
~~~

**Comment by FRidh** [comment]
Oh, thanks. The fix https://github.com/NixOS/nixpkgs/commit/dcbf59d8a780c6d73b1bac10e61ef9d7582c77d7
is in master, staging, and release-17.09 now.

File: pkgs/development/python-modules/oslo-config/default.nix [hunk 1]
```diff
@@ -19,4 +19,7 @@ buildPythonPackage rec {
   postPatch = ''
     substituteInPlace requirements.txt --replace "argparse" ""
   '';
+
+  # Requires a bunch of new packages
+  meta.broken = true;
```
Comment (copumpkin) [inline pkgs/development/python-modules/oslo-config/default.nix]: How did this one become broken when it worked before?

**Comment by FRidh** [comment]
As usual: other package upgrades. Someone should go through the whole Oslo
thing if they want to keep it alive.

File: pkgs/development/python-modules/testtools/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Python 2 only judging from SyntaxError
+#   disabled = isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "15yxz8d70iy1b1x6gd7spvblq0mjxjardl4vnaqasxafzc069zca";
+  };
+
+  propagatedBuildInputs = [ pbr python_mimeparse extras lxml unittest2 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/testtools/default.nix]: @FRidh Do you know why this depends on lxml? I can't find it in https://github.com/testing-cabal/testtools/blob/b4d158ab8aff96c4212d49b3f0d2c3001682c3c9/requirements.txt

Comment (FRidh) [reply pkgs/development/python-modules/testtools/default.nix]: No idea, it was there already. Let's throw it out.

Comment (dotlambda) [reply pkgs/development/python-modules/testtools/default.nix]: I'll try enabling the tests as well

**Pull request merged**